File: messenger_people_java/src/main/java/com/example/messenger_people_java/adapter/StoryAdapter.java
package com.example.messenger_people_java.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.example.messenger_people_java.R;
import com.example.messenger_people_java.model.Story;
import com.google.android.material.imageview.ShapeableImageView;

import java.util.ArrayList;

public class StoryAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private static final int TYPE_ADD_STORY = 1;
    private static final int TYPE_STORY = 0;

    private Context context;
    private ArrayList<Story> items;

    public StoryAdapter(Context context, ArrayList<Story> items) {
        this.context = context;
        this.items = items;
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    @Override
    public int getItemViewType(int position) {
        Story story = items.get(position);
        return story.isAdd() ? TYPE_ADD_STORY : TYPE_STORY;
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());

        if (viewType == TYPE_ADD_STORY) {
            View view = inflater.inflate(R.layout.item_add_story, parent, false);
            return new AddStoryViewHolder(view);
        } else {
            View view = inflater.inflate(R.layout.item_story, parent, false);
            return new StoryViewHolder(view);
        }
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
        Story story = items.get(position);

        if (holder instanceof AddStoryViewHolder) {
            AddStoryViewHolder addStoryViewHolder = (AddStoryViewHolder) holder;
            addStoryViewHolder.iv_story.setImageResource(story.getStory());
            addStoryViewHolder.iv_profile.setImageResource(story.getProfile());
            addStoryViewHolder.tv_fullname.setText(story.getFullname());
        }

        if (holder instanceof StoryViewHolder) {
            StoryViewHolder storyViewHolder = (StoryViewHolder) holder;
            storyViewHolder.iv_story.setImageResource(story.getStory());
            storyViewHolder.iv_profile.setImageResource(story.getProfile());
            storyViewHolder.tv_fullname.setText(story.getFullname());
            storyViewHolder.tv_count.setText(story.getCount());
        }
    }

    public class AddStoryViewHolder extends RecyclerView.ViewHolder {

        ShapeableImageView iv_story;
        ShapeableImageView iv_profile;
        TextView tv_fullname;

        public AddStoryViewHolder(View view) {
            super(view);
            iv_story = view.findViewById(R.id.iv_story);
            iv_profile = view.findViewById(R.id.iv_profile);
            tv_fullname = view.findViewById(R.id.tv_fullname);
        }
    }

    public class StoryViewHolder extends RecyclerView.ViewHolder {
        ShapeableImageView iv_story;
        ShapeableImageView iv_profile;
        TextView tv_fullname;
        TextView tv_count;

        public StoryViewHolder(View view) {
            super(view);
            iv_story = view.findViewById(R.id.iv_story);
            iv_profile = view.findViewById(R.id.iv_profile);
            tv_fullname = view.findViewById(R.id.tv_fullname);
            tv_count = view.findViewById(R.id.tv_count);
        }
    }
}

File: messenger_people_kotlin/src/main/java/com/example/messenger_people_kotlin/adapter/StoryAdapter.kt
package com.example.messenger_people_kotlin.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.messenger_people_kotlin.R
import com.example.messenger_people_kotlin.model.Story
import com.google.android.material.imageview.ShapeableImageView


class StoryAdapter(var context: Context, var items: ArrayList<Story>) :
    RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    private val TYPE_ADD_STORY = 1
    private val TYPE_STORY = 0

    override fun getItemCount(): Int {
        return items.size
    }

    override fun getItemViewType(position: Int): Int {
        var room = items[position]
        if (room.isAdd)
            return TYPE_ADD_STORY
        return TYPE_STORY
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        if(viewType == TYPE_ADD_STORY) {
            val view =
                LayoutInflater.from(parent.context).inflate(R.layout.item_add_story, parent, false)
            return AddStoryViewHolder(view)
        }
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.item_story, parent, false)
        return StoryViewHolder(view)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        val room = items[position]

        if (holder is AddStoryViewHolder) {
            var iv_story = holder.iv_story
            var iv_profile = holder.iv_profile
            var tv_fullname = holder.tv_fullname
            iv_story.setImageResource(room.story)
            iv_profile.setImageResource(room.profile)
            tv_fullname.text = room.fullname
        }

        if (holder is StoryViewHolder) {
            var iv_story = holder.iv_story
            var iv_profile = holder.iv_profile
            var tv_fullname = holder.tv_fullname
            var tv_count = holder.tv_count

            iv_story.setImageResource(room.story)
            iv_profile.setImageResource(room.profile)
            tv_fullname.text = room.fullname
            tv_count.text = room.count
        }
    }

    class AddStoryViewHolder(view: View) : RecyclerView.ViewHolder(view) {

        var iv_story: ShapeableImageView
        var iv_profile: ShapeableImageView
        var tv_fullname: TextView

        init {
            iv_story = view.findViewById(R.id.iv_story)
            iv_profile = view.findViewById(R.id.iv_profile)
            tv_fullname = view.findViewById(R.id.tv_fullname)
        }
    }

    class StoryViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var iv_story: ShapeableImageView
        var iv_profile: ShapeableImageView
        var tv_fullname: TextView
        var tv_count: TextView

        init {
            iv_story = view.findViewById(R.id.iv_story)
            iv_profile = view.findViewById(R.id.iv_profile)
            tv_fullname = view.findViewById(R.id.tv_fullname)
            tv_count = view.findViewById(R.id.tv_count)
        }
    }
}